### PR TITLE
Dashboard: refactor modals for multi manage

### PR DESCRIPTION
**Pull request by iszmais**
https://mantis.ilias.de/view.php?id=42438

This PR refactors the modals on the dashboard to modern form modal dialogs without async confirmation and removes the legacy confirmation form.

### CODE REVIEW

File: Services/Dashboard/Block/classes/class.ilDashboardBlockGUI.php [hunk 1]
```diff
@@ -407,25 +396,100 @@ public function addCommandActions(): void
...
+                [
+                    $this->ui->factory()->messageBox()->confirmation($question),
+                    $this->ui->factory()->messageBox()->info($this->lng->txt('select_one')),
+                ],
+                $items,
+                $this->ctrl->getLinkTargetByClass([ilDashboardGUI::class, $this::class], 'confirmedRemove')
+            )
+                ->withActionButtons([
+                    $this->ui->factory()->button()->standard($this->getRemoveMultipleActionText(), '#')->withOnLoadCode(static fn($id) => "
+                        $id.nextElementSibling.style.display = 'none';
```
Comment (lscharmer) [inline Services/Dashboard/Block/classes/class.ilDashboardBlockGUI.php]: Please extract this code into it's own file.

File: Services/Dashboard/Block/classes/class.ilDashboardBlockGUI.php [hunk 2]
```diff
@@ -407,25 +396,100 @@ public function addCommandActions(): void
...
+                        let modal = $id.closest('.modal-content');
```
Comment (lscharmer) [inline Services/Dashboard/Block/classes/class.ilDashboardBlockGUI.php]:
```suggestion
                        const modal = $id.closest('.modal-content');
```

File: Services/Dashboard/Block/classes/class.ilDashboardBlockGUI.php [hunk 3]
```diff
@@ -407,25 +396,100 @@ public function addCommandActions(): void
...
+                        $id.addEventListener('click', (e) => {
+                            if (modal.querySelector('input:checked') !== null) {
+                                modal.querySelector('.alert-info').hidden = true;
+                                modal.querySelector('.alert-warning').hidden = false;
+                                modal.querySelectorAll('input').forEach((i) => {
+                                    i.hidden = true;
+                                    i.closest('li').hidden = true
```
Comment (lscharmer) [inline Services/Dashboard/Block/classes/class.ilDashboardBlockGUI.php]:
```suggestion
                                    i.closest('li').hidden = true;
```

**Comment by iszmais** [comment]
Hello,

@lscharmer As announced I removed the async rendering of confirmations from the dashboard and integrated the modal into the new UI structure. Pls have a look at the code and the conceptional result.

Due to the usages of the new form, we need to slightly adapt the input of ref_ids for every manageable component within the dashboard. Therefore:
@klees please review `StudyProgramme` and `LearningSequence`
@alex40724 please review `Repository` (RecomendedContent)

Greetings,
@iszmais

File: Services/Dashboard/Block/classes/class.ilDashboardBlockGUI.php [hunk 4]
```diff
@@ -440,8 +504,21 @@ public function executeCommand(): string
                 break;
 
             default:
-                if (method_exists($this, $cmd . 'Object')) {
-                    return $this->{$cmd . 'Object'}();
+                switch ($cmd) {
+                    case 'confirmedRemove':
+                        $form = $this->ui->factory()->input()->container()->form()->standard('', $this->getManageFields())->withRequest($this->http->request());
+                        $ids = [];
+                        foreach ($form->getData() as $group) {
```
Comment (lscharmer) [inline Services/Dashboard/Block/classes/class.ilDashboardBlockGUI.php]: Please have a look if you can drop this `foreach` if you use `->withAdditionalTransformation` in line `486` instead. As this is part of input validation IMO this belongs to the managed fields.

**Review by lscharmer — CHANGES_REQUESTED** [review]

**Review by klees — APPROVED** [review]

**Review by lscharmer — APPROVED** [review]

**Review by alex40724 — APPROVED** [review]

**Pull request merged**